src/txn/txn_ckpt.c

```diff
@@ -767,10 +764,93 @@ __checkpoint_prepare(
 	 */
 	WT_ASSERT(session, session->ckpt_handle_next == 0);
 	WT_WITH_TABLE_READ_LOCK(session, ret = __checkpoint_apply_all(
-	    session, cfg, __wt_checkpoint_get_handles, NULL));
+	    session, cfg, __wt_checkpoint_get_handles));
 	return (ret);
 }
 
+/*
+ * __txn_checkpoint_can_skip --
+ *	Determine whether it's safe to skip taking a checkpoint. This function
+ *	also parses out some configuration options and hands them back to the
+ *	caller - make sure it does that parsing regardless of the result.
+ */
```

Comment:
If it was me, I'd bring this comment down into the function code where it matters, rather than up here? I certainly don't feel strongly though.